static/js/containers/ProfilePage_test.js

```diff
@@ -17,106 +17,76 @@ describe("ProfilePage", function() {
   this.timeout(5000);  // eslint-disable-line no-invalid-this
 
   let listenForActions, renderComponent, helper, patchUserProfileStub;
+  let pageUrlStubs = [
+      '/profile/personal',
+      '/profile/education',
+      '/profile/professional',
+      '/profile/privacy'
+  ];
+  let prevButtonSelector = '.progress-button.previous';
+  let nextButtonSelector = '.progress-button.next';
+
   beforeEach(() => {
     helper = new IntegrationTestHelper();
     listenForActions = helper.listenForActions.bind(helper);
     renderComponent = helper.renderComponent.bind(helper);
-
     patchUserProfileStub = helper.sandbox.stub(api, 'patchUserProfile');
   });
 
   afterEach(() => {
     helper.cleanup();
   });
 
-  it('marks email_optin and filled_out when saving this page', done => {
-    renderComponent("/profile/privacy").then(([, div]) => {
-      let button = div.querySelectorAll("button")[1];
-      assert.equal(button.innerHTML, "I’m Done!");
+  let confirmSaveButtonBehavior = (updatedProfile, pageElements) => {
+    let { div, button } = pageElements;
+    button = button || div.querySelector(nextButtonSelector);
+    patchUserProfileStub.throws("Invalid arguments");
+    patchUserProfileStub.withArgs(SETTINGS.username, updatedProfile).returns(Promise.resolve(updatedProfile));
+    return listenForActions([
+      REQUEST_PATCH_USER_PROFILE,
+      RECEIVE_PATCH_USER_PROFILE_SUCCESS,
+      START_PROFILE_EDIT,
+      UPDATE_PROFILE_VALIDATION
+    ], () => {
+      TestUtils.Simulate.click(button);
+    });
+  };
+
+  it('navigates backward when Previous button is clicked', () => {
+    let pageIndex = 1;
+    renderComponent(pageUrlStubs[pageIndex]).then(([, div]) => {
```

Comment:
I think indexing into `pageUrlStubs` here makes it a lot less clear what this test is actually testing. I think it makes sense to use that pattern when you want to write a test that checks that some behavior occurs across all the pages (something like `pageUrlStubs.map( index => { some.test.code(index) });` ) but if we're only testing what occurs on one page I need to look around unnecessarily to figure out which page we're actually on (`pageUrlStubs[pageIndex] == pageUrlStubs[1] == '/profile/education'`).

Can we change these tests on a single page to just use the string literals? (this test and the next one down).